packages/core/src/actors.ts

```diff
@@ -77,47 +71,51 @@ export type LifecycleSignal = StartSignal | StopSignal;
  */
 
 function isSignal(
-  event: EventObject | LifecycleSignal
-): event is LifecycleSignal {
-  return typeof event.type === 'symbol';
+  eventType: string | Symbol
+): eventType is LifecycleSignalType {
+  return eventType === startSignalType || eventType === stopSignalType;
 }
 
 export function fromCallback<TEvent extends EventObject>(
   invokeCallback: InvokeCallback
 ): Behavior<TEvent, undefined> {
-  let canceled = false;
```

Comment:
Moved these (and others) out of closure; now everything is contained in the behavior and there is no accidental shared state

Reply:
We should add tests for this (I think that I've not seen any new explicit tests about this)

Reply:
What would we be testing? Unsure how to craft a test for this